src/launchpad/service.py

```diff
@@ -0,0 +1,204 @@
+"""Main service orchestrator for Launchpad."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import signal
+from typing import Any, Dict, Union
+
+from .kafka import KafkaConsumer, LaunchpadMessage, get_kafka_config
+from .server import LaunchpadServer, get_server_config
+
+logger = logging.getLogger(__name__)
+
+
+class LaunchpadService:
+    """Main service that orchestrates HTTP server and Kafka consumer."""
+
+    def __init__(self) -> None:
+        self.server: LaunchpadServer | None = None
+        self.kafka_consumer: KafkaConsumer | None = None
+        self._shutdown_event = asyncio.Event()
+        self._tasks: list[Union[asyncio.Task[Any], asyncio.Future[Any]]] = []
+        self._kafka_task: asyncio.Future[Any] | None = None
+
+    async def setup(self) -> None:
+        """Set up the service components."""
+        # Setup HTTP server
+        server_config = get_server_config()
+        self.server = LaunchpadServer(host=server_config["host"], port=server_config["port"])
+
+        # Setup Kafka consumer
+        kafka_config = get_kafka_config()
+        self.kafka_consumer = KafkaConsumer(
+            topics=kafka_config["topics"],
+            group_id=kafka_config["group_id"],
+            bootstrap_servers=kafka_config["bootstrap_servers"],
+            message_handler=self.handle_kafka_message,
+        )
```

Comment:
Will you be able to operate the web and consumer components independently? Being able to do that can help with handling production scale as we can separate the web and consumers into separate k8s deployments.

Reply:
The web server is just for the health check, so it doesnt make sense to separate it. Any scaling needs will solely be around the consumer logic fortunately!